room/runtime/src/main/java/androidx/room/RoomDatabase.java

```diff
@@ -624,6 +625,31 @@ private static boolean isLowRamDevice(@NonNull ActivityManager activityManager)
             return this;
         }
 
+        /**
+         * Configures Room to create and open the database using a pre-packaged gzipped
+         * database file.
+         * <p>
+         * Room does not open the pre-packaged database, instead it copies it into the internal
+         * app database folder, extracts it, and then opens it. The given file must be accessible
+         * and the right permissions must be granted for Room to copy the file.
+         * <p>
+         * The pre-packaged database schema will be validated. It might be best to create your
+         * pre-packaged database schema utilizing the exported schema files generated when
+         * {@link Database#exportSchema()} is enabled.
+         * <p>
+         * This method is not supported for an in memory database {@link Builder}.
+         *
+         * @param databaseFile The database file.
+         *
+         * @return This {@link Builder} instance.
+         */
+        @NonNull
+        public Builder<T> createFromGzippedFile(@NonNull File databaseFile) {
```

Comment:
debated a bit about whether to add this as a new method or just detect based on the filename ending in `.gz`, but opted for this at the end because thought it's cleaner. open to suggestions though.

Reply:
lint doesn't seem happy about this - it wants a method name starting with `set` or `get` instead, but thought this is more in line with what's there.

Reply:
What if instead we do a `createFromInputStream(@NonNull Callable<InputStream>)` such that the user can provide a callable that knows how to create the input stream to be used for the copy and specifically they can then provide us with the `GZIPInputStream` or a `ZipInputStream` (at the right zip entry), etc... and Room would simply copy those bytes and attempt to use it as a DB.

Reply:
done.